README.md

```diff
@@ -913,6 +914,47 @@ java_library(
 
 Any version conflicts or duplicate artifacts will resolved automatically.
 
+## Publishing to External Repositories
+
+In order to publish an artifact from your repo to a maven repository, you
+must first create a `java_export` target. This is similar to a regular
+`java_library`, but allows two additional parameters: the maven coordinates
+and an optional template file to use for the `pom.xml` file.
+
+```python
+# user_project/BUILD
+load("@rules_jvm_external//:defs.bzl", "java_export")
+
+java_export(
+  name = "exported_lib",
+  maven_coordinates = "com.example:project:0.0.1",
+  pom_template = "pom.tmpl",  # You can omit this
+  srcs = glob(["*.java"]),
+  deps = [
+    "//user_project/utils",
+    "@maven//:com_google_guava_guava",
+  ],
+)
+```  
+
+In order to publish the artifact, use `bazel run`:
+
+`bazel run --define "maven_repo=file://$HOME/.m2/repository" //user_project:exported_lib.publish`
```

Comment:
To publish multiple artifacts from the same project (like jersey or jackson split their publications), would that be feasible with a single run target?

Reply:
At the moment, no. In the Selenium project, we have a script that uses a `bazel query` to find the items to deploy, and then calls `bazel run` in a loop. It'd be nice to have a single target for deploying everything….

Reply:
I realize I'm late to the party, but I handle this problem via constructing a parent pom with modules based on a bazel query. We generate pom.xml files for each bazel package via a macro augmenting our `java_library` derivative as well as a standalone macro to declare a maven package comprised of arbitrary artifacts. This allows `mvn` to optimize the artifact upload in cases where we have hundreds of artifacts. I think this approach is feasible to adopt here to allow multiple targets to be uploaded at once.

Reply:
Probably better if we talked about this on the #maven channel in the Bazel Slack. Generating a parent pom would be very useful.